models/firewall.js

```diff
@@ -130,6 +138,36 @@ async function createContentTextChildRuleSideEffect(context) {
       contents: contentsAffected,
     },
   });
+
+  await undoContentsTabcoins(results.rows, createdEvent, context);
+}
+
+async function undoContentsTabcoins(contentRows, createdEvent, context) {
+  for (const contentObject of contentRows) {
+    const tabcoins = await balance.getTotal({
```

Comment:
Aqui seriam evitadas novas idas ao banco de dados se o saldo já for retornado nas funções de firewall.